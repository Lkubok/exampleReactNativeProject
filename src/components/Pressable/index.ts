import Component from './Pressable';

export default Component;
