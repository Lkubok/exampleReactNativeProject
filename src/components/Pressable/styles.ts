import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  pressable: {
    alignItems: 'center',
    justifyContent: 'center',
  },
});
