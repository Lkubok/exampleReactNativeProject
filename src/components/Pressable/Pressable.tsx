import React, { FunctionComponent, ReactNode } from 'react';
import { ViewStyle, StyleProp, TouchableOpacity, StyleSheet } from 'react-native';
import { activeOpacity } from 'utils';

import styles from './styles';

interface Props {
  onPress: () => void;
  disabled?: boolean;
  style?: StyleProp<ViewStyle>;
  children: ReactNode;
}

const Pressable: FunctionComponent<Props> = ({ onPress, style, children, disabled }) => {
  return (
    <TouchableOpacity
      activeOpacity={activeOpacity}
      disabled={disabled}
      onPress={onPress}
      style={StyleSheet.flatten([styles.pressable, style])}
      hitSlop={{ bottom: 2, left: 2, right: 2, top: 2 }}>
      {children}
    </TouchableOpacity>
  );
};

export default Pressable;
