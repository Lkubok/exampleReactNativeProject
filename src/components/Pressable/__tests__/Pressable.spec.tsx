import React from 'react';
import { act, customRender } from 'utils/testUtils';

import Pressable from '../Pressable';

jest.useFakeTimers();

describe('Pressable component', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let props: any;
  beforeEach(() => {
    props = {
      autoCapitalize: 'words',
      blocked: false,
      error: 'testError',
      initialFocus: true,
      isPasswordField: true,
      keyboardType: 'number-pad',
      label: 'test label',
      onBlur: () => {},
      onChange: () => {},
      onChangeText: () => {},
      onFocus: () => {},
      onSubmitEditing: () => {},
      placeholder: 'test placeholder',
      secureTextEntry: true,
      shakeCounter: 5,
      useNativeDriver: true,
      value: 'test value',
    };
  });

  it('should match snapshot when 1 line', () => {
    const snapshot = customRender(<Pressable {...props} />).toJSON();
    // eslint-disable-next-line no-void
    void act(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
