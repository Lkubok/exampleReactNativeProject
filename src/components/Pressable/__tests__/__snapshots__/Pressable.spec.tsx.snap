// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pressable component should match snapshot when 1 line 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    hitSlop={
      Object {
        "bottom": 2,
        "left": 2,
        "right": 2,
        "top": 2,
      }
    }
    onClick={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  />
</RNCSafeAreaProvider>
`;
